### ISSUE TITLE
Consider relaxing canonical/source origin requirement for var subs target URLs

### ISSUE THREAD

**Issue by mkhatib** [ampproject/amphtml#7168]
`amp-form` is going to initially limit variable substitutions to kick in only if the form action is on the canonical origin, otherwise variables will not be substituted. 

This is a tracking bug to consider relaxing that requirement in the future.

- Sprint H2 May: 3 done, 2 left

**Comment by jalajc** [comment]
@mkhatib request you to please explain this further. 
Particularly, by `canonical origin` you mean the exact source URL or source origin?

(In my case I am getting this error `Variable substitutions disabled for non-canonical origin submit action:%s` and I don't find reference anywhere including SO except in code commit TODO #7168.
xhr submitted to: https://<origin>/xhr/ui/xyz.php; 
AMP url: https://<origin>/abcd/amp/
)

**Comment by rudygalfi** [comment]
@aghassemi Can you take a look at this and figure out a course for next steps?

**Comment by aghassemi** [comment]
/to @cvializ 
Let's investigate first why variable substitution was limited to origin ( can't find any info in the design doc and can't really think of a reason), may need to check with reviewers of the PR that added it.

After we figure out why, then we can consider relaxing it.

**Comment by cvializ** [comment]
I found a doc ["URL replacements for links and form inputs in AMP documents"](https://docs.google.com/document/d/1P1drdXrbZGKvMuLWQIAfJ_ceBHCH1tPRnHLnqjupt2I/edit#) (not sure if it's shared publicly) which shows that the restriction was requested, and it aimed to avoid leaking the values to third parties by accident.

In that doc I found [#7344 "Allow variable substitution for whitelisted origins"](https://github.com/ampproject/amphtml/pull/7344) which allows publishers to specify whitelisted origins, but it didn't remove this check from `amp-form`. The issue to document the whitelist is still open (#8132).

It looks like we can remove the restriction in `amp-form` since it's implemented in `url-replacements-impl.js` with the whitelist.
